tests/handlers/tokens_test.py

```diff
@@ -0,0 +1,254 @@
+    app = await create_test_app()
+    client = await aiohttp_client(app)
+
+    r = await client.get(
+        "/auth/tokens/blah", headers={"X-Auth-Request-Token": "foo"}
+    )
+    assert r.status == 401
+    assert r.headers["WWW-Authenticate"]
+
+
+async def test_tokens_handle_get_delete(aiohttp_client: TestClient) -> None:
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair)
+    session_secret = os.urandom(16)
+    app = await create_test_app(keypair, session_secret)
+    client = await aiohttp_client(app)
+
+    ticket = Ticket()
+    scoped_token_payload = create_test_token_payload(
+        scope="exec:test", jti=ticket.as_handle("oauth2_proxy")
+    )
+    scoped_token = jwt.encode(
+        scoped_token_payload,
+        keypair.private_key_as_pem(),
+        algorithm=config.ALGORITHM,
+        headers={"kid": "some-kid"},
+    ).decode()
+    session = Session(
+        token=scoped_token,
+        email="some-user@example.com",
+        user="some-user@example.com",
+        created_at=datetime.now(timezone.utc),
+        expires_on=datetime.now(timezone.utc) + timedelta(days=1),
+    )
+
+    redis_client = app["jwt_authorizer/redis"]
+    session_store = SessionStore("oauth2_proxy", session_secret, redis_client)
+    token_store = TokenStore(redis_client, "uidNumber")
+    pipeline = redis_client.pipeline()
+    session_store.store_session(ticket, session, pipeline)
+    token_store.store_token(scoped_token_payload, pipeline)
+    await pipeline.execute()
+
+    handle = ticket.as_handle("oauth2_proxy")
+    r = await client.get(
+        f"/auth/tokens/{handle}", headers={"X-Auth-Request-Token": token},
+    )
+    assert r.status == 200
+    assert handle in await r.text()
+
+    r = await client.get(
+        "/auth/tokens", headers={"X-Auth-Request-Token": token}
+    )
+    assert r.status == 200
+    body = await r.text()
+    csrf_match = re.search('name="_csrf" value="([^"]+)"', body)
+    assert csrf_match
+    csrf_token = csrf_match.group(1)
+
+    # Deleting without a CSRF token will fail.
+    r = await client.post(
+        f"/auth/tokens/{handle}",
+        headers={"X-Auth-Request-Token": token},
+        data={"method_": "DELETE"},
+    )
+    assert r.status == 403
+
+    # Deleting with a bogus CSRF token will fail.
+    r = await client.post(
+        f"/auth/tokens/{handle}",
+        headers={"X-Auth-Request-Token": token},
+        data={"method_": "DELETE", "_csrf": csrf_token + "xxxx"},
+    )
+    assert r.status == 403
+
+    # Deleting with the correct CSRF will succeed.
+    r = await client.post(
+        f"/auth/tokens/{handle}",
+        headers={"X-Auth-Request-Token": token},
+        data={"method_": "DELETE", "_csrf": csrf_token},
+    )
+    assert r.status == 200
+    body = await r.text()
+    assert f"token with the ticket_id {handle} was deleted" in body
+
+    assert await token_store.get_tokens("1000") == []
```

Comment:
so this 1000 is the uid right?  Maybe stick some of these magic variables together with your test token parts just to make it even more clear with some of these test data things sprinkled about.